### PR TITLE
Ma 340 add event for on bot close

### PR DESCRIPTION
**Pull request by PriyankUpadhyay**
Can listen to botClose events now.
example: 
```java
        ymChat.onBotClose((YMBotEventResponse botEvent) -> {
            Log.d("Example App", "Bot Was closed");
        });
```

### CODE REVIEW

**Comment by kaunteya** [comment]
Please keep this open. I will get this done in iOS tomorrow. Then we can merge it.
We will also have to support this in react-native SDK

File: YMChat/src/main/java/com/yellowmessenger/ymchat/YMChat.java [hunk 1]
```diff
@@ -64,11 +68,17 @@ public void closeBot() {
 
     public void emitEvent(YMBotEventResponse event) {
         if (event != null) {
-            if (listener != null)
-                listener.onSuccess(event);
+            if(botCloseEventListener!= null && event.getCode().equals("bot-closed")){
```
Comment (sukrit-ym) [inline YMChat/src/main/java/com/yellowmessenger/ymchat/YMChat.java:71]: can event.getCode() be null? If yes, lets put a null check for this as well.

File: YMChat/src/main/java/com/yellowmessenger/ymchat/BotWebView.java [hunk 2]
```diff
@@ -177,6 +177,7 @@ public void onCreate(Bundle savedInstanceState) {
                 case "close-bot":
                     closeBot();
                     this.finish();
+                    YMChat.getInstance().emitEvent(new YMBotEventResponse("bot-closed", ""));
```
Comment (sukrit-ym) [inline YMChat/src/main/java/com/yellowmessenger/ymchat/BotWebView.java:180]: Put this line before calling this.finish(), otherwise it may not get called.

Comment (PriyankUpadhyay) [reply YMChat/src/main/java/com/yellowmessenger/ymchat/BotWebView.java:179]: Done

File: YMChat/src/main/java/com/yellowmessenger/ymchat/YMChat.java (hunk 1 above)
Comment (PriyankUpadhyay) [reply YMChat/src/main/java/com/yellowmessenger/ymchat/YMChat.java]: added null check

**Review by sukrit-ym — APPROVED** [review]

**Comment by kaunteya** [comment]
@sukrit-ym we should call bot-close when user taps on close button

**Pull request merged**